components/script/dom/range.rs

```diff
@@ -343,7 +343,7 @@ impl RangeMethods for Range {
         } else {
             // Step 3.
             self.set_start(node, offset);
-            if !(self.start() <= self.end()) {
```

Comment:
I think this hasn't been changed up until now because in this case `>` is actually different from `<=` for some reason. I believe that the right thing to do here is to preserve the original conditional and put an `allow` directive with a comment.

Reply:
I checked it again and you are totally right. BoundaryPoint can be uncomparable. 
Clippy suggests to use an explicit reminder about this case 
```
let _not_less_or_equal = match a.partial_cmp(&b) {
    None | Some(Ordering::Greater) => true,
    _ => false,
};
```

What do you think is better? allow or match?

Reply:
I think `allow` is clearer here. Thanks!